test/RpcEvmCompatibility.js

```diff
@@ -54,14 +54,46 @@ describe('EVM RPC Compatibility Test', () => {
         const newValue = 100;
         const receipt = await rpcCompatibilityContract.updateValue(newValue).then((tx) => tx.wait());
         const blockNumber = hexValue(receipt.blockNumber);
+        const logs = [];
 
-        const filter = {
+        let filter = {
             fromBlock: blockNumber,
             toBlock: blockNumber,
         };
-        const logs = await provider.send('eth_getLogs', [filter]);
-        expect(logs).to.be.an('array');
-        expect(logs.length).to.be.greaterThan(0);
+        let log = await provider.send('eth_getLogs', [filter]);
+        logs.push(log);
+        filter = {
+            fromBlock: blockNumber,
+            toBlock: 'latest',
+        };
+        log = await provider.send('eth_getLogs', [filter]);
+        logs.push(log);
+        filter = {
+            fromBlock: blockNumber,
+            toBlock: 'pending',
+        };
+        log = await provider.send('eth_getLogs', [filter]);
+        logs.push(log);
+
+        if (network.name.toLowerCase() === 'ethereum') {
+            filter = {
+                fromBlock: blockNumber,
+                toBlock: 'safe',
+            };
+            log = await provider.send('eth_getLogs', [filter]);
+            logs.push(log);
+            filter = {
+                fromBlock: blockNumber,
+                toBlock: 'finalized',
+            };
+            log = await provider.send('eth_getLogs', [filter]);
+            logs.push(log);
+        }
+
+        logs.forEach((log) => {
+            expect(log).to.be.an('array');
+            expect(log.length).to.be.greaterThan(0);
```

Comment:
at least for the latest block filter test, should look for the event emitted in our tx